client/asset/btc/btc.go

```diff
@@ -1301,6 +1310,12 @@ func (btc *ExchangeWallet) FindRedemption(ctx context.Context, coinID dex.Bytes)
 		delete(btc.findRedemptionQueue, contractOutpoint)
 	}
 	btc.findRedemptionMtx.Unlock()
+
+	// Don't abandon the goroutines even if context canceled.
+	// findRedemptionsInTx will return when it fails to find the assigned
+	// contract outpoint in the findRedemptionQueue.
+	wg.Wait()
```

Comment:
Hmm. I think you're right about modifying the `findRedemptionsInMempool` and `findRedemptionsInBlockRange` methods receive and use the context. It turns out I didn't do this correctly earlier - canceling the context does not prevent the started goroutine(s) from checking all checkables before returning; although any result returned will be ignored.

I still think they should be goroutines though, to prevent blocking the calling thread; unless we'll now advise callers to call the FindRedemption from a goroutine where a blocking call is undesirable (which is everywhere else outside tests).

Reply:
This does need refactoring, but we need to get tests passing again.  The core of the issue we keep hitting (e.g. just now in https://github.com/decred/dcrdex/pull/688/checks?check_run_id=1144164460#step:5:198) is that a cancelled/timed-out context does not wait for the queued search to be removed from the queue map (and any error or result may get pulled by subsequent calls depending on timing).

Reply:
> I still think they should be goroutines though, to prevent blocking the calling thread; unless we'll now advise callers to call the FindRedemption from a goroutine where a blocking call is undesirable (which is everywhere else outside tests).

I wanted to comment on this.  `FindRedemption` is presently blocking already (waits to receive on `resultChan`), so I'm a bit confused about the comment.  My understanding of the current design is that the goroutines themselves don't do a lot or take long, *but* the search gets queued to check on new blocks if the redeem isn't found right away, and by `delete(btc.findRedemptionQueue, contractOutpoint)`, this is effectively canceling the ongoing search if not the immediate search.  I'm fine with that for now... just as long as context cancellation lets the immediate search complete so the result/err doesn't bleed into a subsequent call for the same outpoint.

Reply:
Argh. You're right. My brain was recalling from before when I made FindRedemption start goroutines **without** waiting on them. I think I'm gonna get on that refactor right after this is merged, really doesn't make sense to have FindRedemption start a goroutine anymore.

Reply:
Sounds good, although 677 is top prio ofc.